lib/symboldatabase.cpp

```diff
@@ -4202,8 +4204,39 @@ static const Token* skipPointersAndQualifiers(const Token* tok)
     return tok;
 }
 
+static const Token* skipTypeOrFunctionIdentifier(const Token* tok)
+{
+    if (tok == nullptr)
+        return nullptr;
+    while (Token::Match(tok->next(), "::|<"))
+    {
+        if (tok->next()->str() == "::")
+            tok = tok->tokAt(2);
+        else
+            tok = tok->linkAt(1);
+        if (tok == nullptr)
+            return nullptr;
+    }
+    return tok->next();
+}
+
+//Could this token be the first token of a call to a function instead of the declaration of a variable
+bool isFunctionCallInsteadOfVariableDeclaration(const Token* tok, const Scope& scope, const SymbolDatabase& db)
+{
+    auto start = tok;
+    tok = skipTypeOrFunctionIdentifier(tok);
+    if (tok == nullptr)
+        return false; //It's probably neither but it's not a function call
+    if (tok->str() != "(") //X y isn't a function call
+        return false;
+    return db.findFunction(start, &scope) != nullptr;
```

Comment:
I removed the reference to the AST as per your advice. It was just an optimization when the AST does exist so I think it would be fine whether or not it exists under the current design but it would be nice if the design was improved so that at any stage in the process information either was or was not available rather than being sort of available sometimes. With this in mind I don't want to add code that implies things existing that may not exist.